examples/standalone/light_control/light_control.cc

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <chrono>
+#include <iostream>
+
+#include <ignition/gazebo/components/Light.hh>
+#include <ignition/gazebo/components/LightCmd.hh>
+#include <ignition/msgs.hh>
+#include <ignition/transport.hh>
+
+using namespace std::chrono_literals;
+
+// Create a transport node.
+ignition::transport::Node node;
+
+bool result;
+// timeout used for services
+constexpr unsigned int timeout = 5000;
+
+std::mt19937 twister(std::time(0));
+std::uniform_real_distribution<double> distr(0, 1000);
+constexpr double epsilon = 0.1;
+
+float directionX = 0.5;
+float directionY = 0.2;
+float directionZ = -0.9;
+
+void createLight()
+{
+  ignition::msgs::Boolean rep;
+//! [create light]
+  ignition::msgs::EntityFactory entityFactoryRequest;
+
+  entityFactoryRequest.mutable_light()->set_name("point");
+  entityFactoryRequest.mutable_light()->set_range(4);
+  entityFactoryRequest.mutable_light()->set_attenuation_linear(0.5);
+  entityFactoryRequest.mutable_light()->set_attenuation_constant(0.2);
+  entityFactoryRequest.mutable_light()->set_attenuation_quadratic(0.01);
+  entityFactoryRequest.mutable_light()->set_cast_shadows(false);
+  entityFactoryRequest.mutable_light()->set_type(ignition::msgs::Light::POINT);
+  ignition::msgs::Set(
+    entityFactoryRequest.mutable_light()->mutable_direction(),
+    ignition::math::Vector3d(directionX, directionY, directionZ));
```

Comment:
just a note that point lights ignore the direction field. Only affects spot / directional lights